src/shogun/base/progress.h

```diff
@@ -58,6 +58,10 @@
 
 namespace shogun
 {
+//#define progress(...)
+// Progress(std::string(this->get_name())+"::"+std::string(__FUNCTION__),
+//__VA_ARGS__)
+#define progress(...) Progress(std::string(__FUNCTION__), __VA_ARGS__)
```

Comment:
As a common practice, shouldn't we name macros with uppercase names (so that we know it is a macro and not a function/method call)? It may be a pain to rename everything again though.